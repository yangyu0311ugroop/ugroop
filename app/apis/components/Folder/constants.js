export const FOLDER_FETCH_TYPES = {
  PAGINATION: 'pagination',
  SORT: 'sort',
  INITIAL_FETCH: 'initialFetch',
};
