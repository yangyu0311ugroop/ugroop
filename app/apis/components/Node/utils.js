import lodash from 'lodash';
import { compose } from 'redux';
import {
  FOLDER,
  TAB_GALLERY,
  TAB_OTHER,
  TAB_TIMELINE,
  TEMPLATE,
} from 'utils/modelConstants';
import upsertHelpers, { ARRAY_MODE } from 'utils/helpers/upsertStore';
import { DATASTORE_UTILS } from 'datastore';
import { FILE_STORE_SCHEMAS } from 'datastore/fileStore/schema';
import { normalize } from 'normalizr';

export const createTemplateData = data => ({
  ...data,
  id: data.cloneId,
  people: [data.createdBy],
  children: [],
  content: data.content,
  customData: {
    shortDescription: data.shortDescription || '',
    duration: data.duration || 0,
  },
});

export const createNodeData = (type, data) => {
  switch (type) {
    case TEMPLATE: {
      return createTemplateData(data);
    }

    case FOLDER: {
      return {
        ...data,
        id: data.cloneId,
      };
    }

    default:
      return data;
  }
};

export const upsertCalculatedTimes = props => nodes => {
  props.resaga.setValue({
    nodes: upsertHelpers.deepMerge(nodes, ARRAY_MODE.SET),
  });
};

export const upsertActivityData = (node, payload) => {
  const photo = lodash.get(payload, 'node.customData.photo', undefined)
    ? {
        metaInfo: lodash.get(payload, 'node.customData.metaInfo', undefined),
        content: lodash.get(payload, 'node.customData.photo', undefined),
      }
    : null;
  const photoContent = lodash.get(photo, 'content', '');
  const customData = node.customData;
  const attachmentId = lodash.get(payload, 'node.attachment', null);
  const attachment = lodash.get(payload, 'node.customData.attachment', {});
  const attachmentDesc = lodash.get(
    payload,
    'node.customData.attachment.description',
    '',
  );

  const normalizedPhoto = photo
    ? normalize(photo, FILE_STORE_SCHEMAS.templatePhoto)
    : {};

  const attachmentOperations = attachmentId
    ? [
        DATASTORE_UTILS.upsertObject({ [attachmentId]: attachment }), // upsert attachment
        DATASTORE_UTILS.updateAttribute(
          `${attachmentId}.description`,
          attachmentDesc,
        ), // manually update description attribute
      ]
    : [];

  const photoOperations = photo
    ? [
        DATASTORE_UTILS.upsertObject(normalizedPhoto.entities.photo),
        DATASTORE_UTILS.removeObjectById(node.oldPhotoId),
      ]
    : [];

  const nodePhotoOperations = photo
    ? [
        DATASTORE_UTILS.upsertArray(`${payload.nodeId}.photos`, [photoContent]), // manually upsert photo
        DATASTORE_UTILS.upsertObject({
          [node.id]: { ...node, customData: { ...customData } },
        }),
        DATASTORE_UTILS.removeItemsArray(
          `${payload.nodeId}.photos`,
          node.oldPhotoId,
        ),
      ]
    : [];

  return {
    files: compose(...photoOperations),
    attachments: compose(...attachmentOperations),
    nodes: compose(
      DATASTORE_UTILS.upsertObject({
        [node.id]: { ...node, customData: { ...customData } },
      }),
      ...nodePhotoOperations,
    ),
  };
};

const getGalleryId = (template = {}) => {
  const { children } = template;
  if (Array.isArray(children)) {
    for (let i = 0; i < children.length; i += 1) {
      const { id, type } = children[i];

      if (type === TAB_GALLERY) {
        return id;
      }
    }
  }

  return -1;
};

const getPeopleTabId = (template = {}) => {
  const { children } = template;

  if (Array.isArray(children)) {
    for (let i = 0; i < children.length; i += 1) {
      const { id, type, customData } = children[i];

      if (type === TAB_OTHER && customData.subtype === 'tabpeople') {
        return id;
      }
    }
  }

  return -1;
};

const getTimelineId = (template = {}) => {
  const { children } = template;
  if (Array.isArray(children)) {
    for (let i = 0; i < children.length; i += 1) {
      const { id, type } = children[i];

      if (type === TAB_TIMELINE) {
        return id;
      }
    }
  }

  return -1;
};

export const NODE_API_UTILS = {
  getGalleryId,
  getTimelineId,
  getPeopleTabId,
  createTemplateData,
  createNodeData,
  upsertCalculatedTimes,
  upsertActivityData,
};
