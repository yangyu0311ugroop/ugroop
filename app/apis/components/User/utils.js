import { normalize } from 'normalizr';
import { compose } from 'redux';
import UserStoreSchema from 'datastore/userStore/schema';
import { DATASTORE_UTILS } from 'datastore';
import orderBy from 'lodash/orderBy';
import { LOGIC_HELPERS } from 'utils/helpers/logic';
import _ from 'lodash';

export const retainValue = store => store;

const normaliseRoles = (result, { userId }) => {
  // This filtering could be done in the backend
  // to lessen the process the front-end is doing
  const filteredTours = orderBy(
    result.tours.filter(tour => tour.role !== 'owner'),
    'date',
    'desc',
  );
  const filteredOrgs = orderBy(result.orgs, 'createdAt', 'desc');
  const orgs = normalize(filteredOrgs, UserStoreSchema.organisation);
  const tours = normalize(filteredTours, UserStoreSchema.tour);

  const orgUsers = LOGIC_HELPERS.ifElse(
    !orgs.result.length > 0,
    retainValue,
    DATASTORE_UTILS.upsertObject(orgs.entities.organisation),
  );
  const userNodes = LOGIC_HELPERS.ifElse(
    !tours.result.length > 0,
    retainValue,
    DATASTORE_UTILS.upsertObject(tours.entities.tour),
  );
  const setValue = {
    person: compose(
      DATASTORE_UTILS.upsertArray(`${userId}.orgUsers`, orgs.result),
      DATASTORE_UTILS.upsertArray(`${userId}.userNodes`, tours.result),
    ),
    orgUsers,
    userNodes,
    raw: result,
  };
  return setValue;
};

const addUser = result => {
  const normalizedData = normalize(result.data, UserStoreSchema.plans);
  const products = _.mapValues(normalizedData.entities.plans, o => o.planIds);
  const plans = _.reduce(
    _.mapValues(normalizedData.entities.plans, o => o.plans),
    (all, value) => _.merge(all, value),
    {},
  );
  return {
    plans,
    products,
  };
};

export const USER_APIS_UTILS = {
  normaliseRoles,
  addUser,
};
