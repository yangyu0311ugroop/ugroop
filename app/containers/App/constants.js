/*
 * AppConstants
 * Each action has a corresponding type, which the reducer knows and picks up on.
 * To avoid weird typos between the reducer and the actions, we save them as
 * constants here. We prefix them with 'yourproject/YourComponent' so we avoid
 * reducers accidentally picking up actions they shouldn't.
 *
 * Follow this format:
 * export const YOUR_ACTION_CONSTANT = 'yourproject/YourContainer/YOUR_ACTION_CONSTANT';
 */

export const DEFAULT_LOCALE = 'en';
export const UGROOPSTORE_KEY_PREFIX = 'UGroopStorage:';
export const USERLOGOUT = 'USER_LOGOUT';
export const USERLOGIN = 'USER_LOGIN';
export const REFRESH_TOKEN_EXPIRY = 'REFRESH_TOKEN_EXPIRY';
export const RESET_GLOBAL_ERROR = 'RESET_GLOBAL_ERROR';
