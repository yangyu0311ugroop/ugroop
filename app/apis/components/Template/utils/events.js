/**
 * Created by stephenkarpinskyj on 12/4/18.
 */
import { DATASTORE_UTILS } from 'datastore';
import { EVENT_PATHS } from 'datastore/eventStore/constants';
import { EVENT_STORE_HELPERS } from 'datastore/eventStore/helpers';
import { NODE_BATCH_ACTIONS, NODE_PATHS } from 'datastore/nodeStore/constants';
import { NODE_STORE_HELPERS } from 'datastore/nodeStore/helpers';
import dotProp from 'dot-prop-immutable';
import { EVENT_UTILS } from 'utils/events';
import upsertHelpers, { ARRAY_MODE } from 'utils/helpers/upsertStore';

export const convertToBatchCreate = model => {
  let m = model;

  const batchCreatePath = EVENT_STORE_HELPERS.pathToEventInputName(
    EVENT_PATHS.batchCreate,
  );
  const batchCreate = dotProp.get(model, batchCreatePath, false);
  m = dotProp.delete(m, batchCreatePath);

  if (batchCreate) {
    const dayRangePath = EVENT_STORE_HELPERS.pathToEventInputName(
      EVENT_PATHS.tempDayRange,
    );
    const dayRange = dotProp.get(model, dayRangePath, []);
    m = dotProp.delete(m, dayRangePath);

    const tempStartDayPath = NODE_STORE_HELPERS.pathToNodeInputName(
      NODE_PATHS.tempStartDay,
    );

    return dayRange.map(id => {
      let currM = m;
      currM = dotProp.set(currM, tempStartDayPath, `${id}`);
      // currM = dotProp.set(currM, tempEndDayPath, DEFAULT_DURATION);
      return currM;
    });
  }

  return [m];
};

export const injectNodeType = (node, data) => {
  const model = { node, data };
  const type = dotProp.get(
    model,
    EVENT_STORE_HELPERS.pathToEventInputName(EVENT_PATHS.type),
    null,
  );
  const subtype = dotProp.get(
    model,
    EVENT_STORE_HELPERS.pathToEventInputName(EVENT_PATHS.subtype),
    null,
  );
  return {
    ...node,
    type: EVENT_UTILS.dataTypeToNodeType(type, subtype),
  };
};

export const injectDataId = (data, dataId) => ({
  ...data,
  id: dataId,
});

export const wrapInBatchOperations = (node, id, parentNodeId, dataId) => {
  const updateEvent = {
    node,
    id,
    action: NODE_BATCH_ACTIONS.UPDATE_EVENT,
  };

  const moveInside = parentNodeId
    ? {
        parentId: parentNodeId,
        toBeMovedId: id,
        action: NODE_BATCH_ACTIONS.MOVE_INSIDE,
      }
    : null;

  const operations = [];
  operations.push(updateEvent);
  if (moveInside) operations.push(moveInside);
  return { operations, dataId };
};

const upsertEvents = props => ({
  events,
  flightBookings,
  attachments,
  activityDetail,
}) => {
  const obj = {};
  if (activityDetail) {
    obj.activityDetail = DATASTORE_UTILS.upsertObject(activityDetail);
  }
  if (events) {
    obj.events = DATASTORE_UTILS.upsertObject(events);
  }
  if (attachments) {
    obj.eventAttachments = DATASTORE_UTILS.upsertObject(attachments);
  }
  if (flightBookings) {
    obj.flightBookings = DATASTORE_UTILS.upsertObject(flightBookings);
  }
  obj.flightBookingIds = upsertHelpers.array(
    DATASTORE_UTILS.getObjectIds(flightBookings),
    ARRAY_MODE.SET,
  );
  props.resaga.setValue(obj);
};

const upsertEvent = props => params => {
  const { node, calculatedNodes, events, activityDetail, attachments } = params;

  const obj = { amountLastUpdated: Date.now() };

  if (node) {
    obj.nodes = upsertHelpers.deepMerge(node, ARRAY_MODE.APPEND);
  }
  if (attachments) {
    obj.eventAttachments = upsertHelpers.deepMerge(
      attachments,
      ARRAY_MODE.APPEND,
    );
  }
  if (calculatedNodes) {
    obj.calculatedNodes = upsertHelpers.deepMerge(
      calculatedNodes,
      ARRAY_MODE.APPEND,
    );
  }
  if (events) obj.events = upsertHelpers.deepMerge(events, ARRAY_MODE.SET);
  if (activityDetail)
    obj.activityDetail = upsertHelpers.deepMerge(
      activityDetail,
      ARRAY_MODE.SET,
    );
  props.resaga.setValue(obj);
};

const deleteEvent = props => (_, { templateId, id }) => {
  props.resaga.setValue({
    amountLastUpdated: Date.now(),
    nodes: nodes => ({ ...nodes, [id]: undefined }),
    calculatedNodes: DATASTORE_UTILS.removeItemsArray(
      `${templateId}.calculated.events`,
      id,
    ),
  });
};

const upsertFlightBooking = props => ({ flightBookings }) => {
  const obj = { amountLastUpdated: Date.now() };
  if (flightBookings) {
    obj.flightBookings = upsertHelpers.deepMerge(
      flightBookings,
      ARRAY_MODE.SET,
    );
    obj.flightBookingIds = upsertHelpers.deepMerge(
      DATASTORE_UTILS.getObjectIds(flightBookings),
      ARRAY_MODE.APPEND,
    );
  }
  props.resaga.setValue(obj);
};

const deleteFlightBooking = props => (_, { dataId }) => {
  const obj = {
    flightBookings: DATASTORE_UTILS.removeObjectById(dataId),
    flightBookingIds: DATASTORE_UTILS.removeItemsInArray(dataId),
    amountLastUpdated: Date.now(),
  };
  props.resaga.setValue(obj);
};

export const TEMPLATE_API_EVENT_UTILS = {
  convertToBatchCreate,
  injectNodeType,
  injectDataId,
  wrapInBatchOperations,
  upsertEvents,
  upsertEvent,
  deleteEvent,
  upsertFlightBooking,
  deleteFlightBooking,
};
