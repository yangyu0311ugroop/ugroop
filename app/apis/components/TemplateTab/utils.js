import { NODE_SCHEMA } from 'datastore/nodeStore/schema';
import { compose } from 'redux';
import { normalize } from 'normalizr';
import get from 'lodash/get';
import assign from 'lodash/assign';
import { DATASTORE_UTILS } from 'datastore';
import TabHelper from 'datastore/templateManagementStore/helpers/tabs';
import { TAB_GALLERY } from 'utils/modelConstants';

export const TEMPLATE_TAB_UTILS = {
  addTab: result => {
    const normalizedNode = normalize(result, NODE_SCHEMA.singleNode);
    return {
      nodes: DATASTORE_UTILS.upsertObject(normalizedNode.entities.node),
    };
  },
  addMultipleNode: (result = []) => {
    const nodes = [];
    const tempResult = [...result];

    tempResult.forEach(node => {
      const processedResult = TabHelper.convertChildrenToArray(
        node.type,
        node.children,
      );
      const normalizedNode = normalize(processedResult, NODE_SCHEMA.node);
      nodes.push(normalizedNode);
    });

    // replace tab children with ids rather than objects
    const transformedResult = result
      .map((node, index) => {
        const tempNode = node;
        tempNode.children = nodes[index].result;
        tempNode.index = index;
        return tempNode;
      })
      .filter(node => node.type !== TAB_GALLERY);

    const photos = nodes.reduce(
      (acc, { entities }) => assign(acc, entities.photo),
      {},
    );

    const eventNodes = nodes.reduce(
      (acc, { entities }) => assign(acc, entities.eventNodes),
      {},
    );

    const nonTabNodes = nodes.reduce(
      (acc, { entities }) => assign(acc, entities.node),
      {},
    );

    const parentArr = result.filter(item => !!item.parentNodeId);

    const tabNodes = normalize(transformedResult, NODE_SCHEMA.node);
    let templateId = get(result, '0.parentNodeId');
    if (!templateId) {
      templateId = get(parentArr, '0.parentNodeId');
    }

    return {
      nodes: compose(
        DATASTORE_UTILS.upsertObject(nonTabNodes),
        DATASTORE_UTILS.upsertObject(eventNodes),
        DATASTORE_UTILS.upsertObject(tabNodes.entities.node),
        DATASTORE_UTILS.upsertArray(
          `${templateId}.calculated.events`,
          DATASTORE_UTILS.getObjectIds(eventNodes),
        ),
      ),
      files: DATASTORE_UTILS.upsertObject(photos),
      result: tabNodes.result,
    };
  },
};
