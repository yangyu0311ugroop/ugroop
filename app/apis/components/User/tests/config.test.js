import {
  CREATE_ORG_USER,
  GET_INVITATIONS,
  GET_ROLES,
  USER_API,
  GET_RECENT_ACTIVITY,
  HIDE_RECENT_ACTIVITY,
  CREATE_USER_VIA_INVITE,
  SIGN_IN,
  SIGN_OUT,
  FORGET_PWD,
  RESET_PWD,
  RESEND_SIGNUP,
  ME,
  PERSON_SYNC,
  CHANGE_PWD,
  GET_ORGANISATION_INVITATIONS,
  REGISTER_DEVICE,
  UNREGISTER_DEVICE,
  CREATE_PERSONAL_REGISTRATION,
  GET_USER_PREFERENCE,
  UPDATE_USER_PREFERENCE,
  GET_USER_RELATED_TEMPLATES,
  GET_USER_NODE,
  GET_TRANSFER_NODE_LIST,
  GET_PERSONAL_PREFERENCES,
  UPSERT_PERSONAL_PREFERENCES,
} from 'apis/constants';
import { PENDING, RECEIVED } from 'datastore/invitationStore/constants';
import { requests } from 'utils/request';
import { AwsApi } from 'utils/cognito';
import {
  CONFIG,
  beforeCreateUser,
  convertToOrgKey,
  trimOrgName,
  errorProcesser,
} from '../config';

/**
 * RANDOM STRING GENERATOR
 *
 * Info:      http://stackoverflow.com/a/27872144/383904
 * Use:       randomString(length [,"A"] [,"N"] );
 * Default:   return a random alpha-numeric string
 * Arguments: If you use the optional "A", "N" flags:
 *            "A" (Alpha flag)   return random a-Z string
 *            "N" (Numeric flag) return random 0-9 string
 */
function randomString(len) {
  let str = ''; // String result
  for (let i = 0; i < len; i += 1) {
    // Loop `len` times
    let rand = Math.floor(Math.random() * 62); // random: 0..61
    // eslint-disable-next-line no-multi-assign
    const charCode = (rand += rand > 9 ? (rand < 36 ? 55 : 61) : 48); // eslint-disable-line no-nested-ternary
    str += String.fromCharCode(charCode); // add Character to str
  }
  return str; // After all loops are done, return the concatenated string
}

describe('User/config.js', () => {
  afterEach(() => jest.clearAllMocks());

  describe('Smoke Test', () => {
    it('should exists', () => {
      expect(typeof CONFIG).toBe('object');
    });
  });

  describe('name', () => {
    it('should exists', () => {
      expect(CONFIG.name).toBe(USER_API);
    });
  });

  describe('convertToOrgKey()', () => {
    it('should convert correct Org Key', () => {
      const name =
        'This is My Org Name I - think this is going to work indeed!!!!@#$%^&';
      const convertKey = convertToOrgKey(name);
      expect(convertKey).toEqual(
        'thisismyorgnameithinkthisisgoingtoworkindeed',
      );
    });
    it('should convert correct Org Key', () => {
      const name = 'uGroop';
      const convertKey = convertToOrgKey(name);
      expect(convertKey).toEqual('ugroop');
    });
    it('should genereate 63 chars', () => {
      const name = randomString(255);
      const convertKey = convertToOrgKey(name);
      expect(convertKey.length).toEqual(59);
    });
  });

  describe('trimOrgName()', () => {
    it('should convert correct Org Key', () => {
      const name =
        'This is My Org Name I - think this is working to work indeed!!!!@#$%^&';
      const trimOrg = trimOrgName(name, name);
      expect(trimOrg).toEqual(name);
    });
  });

  describe('beforeCreateUser()', () => {
    it('should return correct Org Key', () => {
      const key = beforeCreateUser({ orgName: 'a', orgAddress: 'b' });
      expect(key).toEqual({
        email: undefined,
        givenName: undefined,
        location: { address: 'b' },
        orgName: 'a',
        orgNameKey: 'org-a-b',
        password: undefined,
        surname: undefined,
      });
    });
  });

  describe('errorProcesser()', () => {
    it('should return correct data', () => {
      const data = errorProcesser();
      expect(data).toEqual({ error: undefined, msg: undefined });
    });
    it('should return correct data', () => {
      const error = { response: { error: { headers: { error: 'abcd' } } } };
      const data = errorProcesser(error);
      expect(data).toEqual({ error, msg: 'abcd' });
    });
    it('should return correct data', () => {
      const error = { response: { error: { message: 'abcd' } } };
      const data = errorProcesser(error);
      expect(data).toEqual({ error, msg: 'abcd' });
    });
    it('should return correct data', () => {
      const error = { message: 'abcd' };
      const data = errorProcesser(error);
      expect(data).toEqual({ error, msg: 'abcd' });
    });
    it('should return correct data', () => {
      const error = { somethingelse: 'abcd' };
      const data = errorProcesser(error);
      expect(data).toEqual({ error, msg: error });
    });
  });

  describe('#requests', () => {
    it('still matches snapshot', () => {
      requests.fetchWithAuthorisation = jest.fn();

      CONFIG.requests[GET_INVITATIONS]({});
      CONFIG.requests[GET_INVITATIONS]({ show: RECEIVED, status: PENDING });

      expect(requests.fetchWithAuthorisation).toBeCalled();
      expect(requests.fetchWithAuthorisation.mock.calls).toMatchSnapshot();
    });

    describe('CREATE_ORG_USER', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[CREATE_ORG_USER]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithURL = jest.fn();
        const data = {
          orgName: 'avcd',
          orgAddress: 'a',
          firstName: 'v',
          lastName: 'c',
          email: 'ads',
          password: 'ssd',
          namekey: 's',
        };
        CONFIG.requests[CREATE_ORG_USER](data);

        expect(requests.fetchWithURL).toBeCalledWith(
          'post',
          `/${USER_API}`,
          beforeCreateUser(data),
        );
      });
    });

    describe('CREATE_USER_VIA_INVITE', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[CREATE_USER_VIA_INVITE]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithURL = jest.fn();
        const data = {
          orgName: 'avcd',
          orgAddress: 'a',
          firstName: 'v',
          lastName: 'c',
          email: 'ads',
          password: 'ssd',
          namekey: 's',
        };
        CONFIG.requests[CREATE_USER_VIA_INVITE](data);

        expect(requests.fetchWithURL).toBeCalledWith(
          'post',
          `/${USER_API}/createUserViaInvite`,
          data,
        );
      });
    });
    describe('CREATE_PERSONAL_REGISTRATION', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[CREATE_PERSONAL_REGISTRATION]).toBe(
          'function',
        );
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithURL = jest.fn();
        const data = {
          givenName: 'v',
          surname: 'c',
          email: 'ads',
          password: 'ssd',
        };
        CONFIG.requests[CREATE_PERSONAL_REGISTRATION](data);

        expect(requests.fetchWithURL).toBeCalledWith(
          'post',
          `/${USER_API}/createSimpleUser`,
          data,
        );
      });
    });

    describe('ME', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[ME]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[ME]();

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/me`,
        );
      });
    });

    describe('REGISTER_DEVICE', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[REGISTER_DEVICE]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[REGISTER_DEVICE]({ id: 1, data: { token: 'token' } });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'post',
          `/${USER_API}/1/registerDevice`,
          { token: 'token' },
        );
      });
    });

    describe('UNREGISTER_DEVICE', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[UNREGISTER_DEVICE]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[UNREGISTER_DEVICE]({ id: 1, data: { token: 'token' } });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'post',
          `/${USER_API}/1/unregisterDevice`,
          { token: 'token' },
        );
      });
    });
    describe('UPDATE_USER_PREFERENCE', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[UPDATE_USER_PREFERENCE]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[UPDATE_USER_PREFERENCE]({
          id: 1,
          data: { token: 'token' },
        });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'patch',
          `/${USER_API}/1/upsertUserPreference`,
          { token: 'token' },
        );
      });
    });
    describe('GET_USER_PREFERENCE', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[GET_USER_PREFERENCE]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[GET_USER_PREFERENCE]({
          id: 1,
          data: { token: 'token' },
        });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/1/userPreference`,
        );
      });
      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[GET_USER_PREFERENCE]({
          id: 1,
          data: { token: 'token' },
        });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/1/userPreference`,
        );
      });
    });
    describe('GET_USER_NODE', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[GET_USER_NODE]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[GET_USER_NODE]();

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/me/userNode`,
        );
      });
    });
    describe('GET_TRANSFER_NODE_LIST', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[GET_TRANSFER_NODE_LIST]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[GET_TRANSFER_NODE_LIST]({});

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/me/transfers?show=default&status=pending`,
        );
      });
    });
    describe('GET_PERSONAL_PREFERENCES', () => {
      it('should exist', () => {
        expect(typeof CONFIG.requests[GET_PERSONAL_PREFERENCES]).toBe(
          'function',
        );
      });
      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[GET_PERSONAL_PREFERENCES]({ id: 1 });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/1/Preferences/personal`,
        );
      });
    });
    describe('UPSERT_PERSONAL_PREFERENCES', () => {
      it('should exist', () => {
        expect(typeof CONFIG.requests[UPSERT_PERSONAL_PREFERENCES]).toBe(
          'function',
        );
      });
      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[UPSERT_PERSONAL_PREFERENCES]({ id: 1, data: 1 });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'patch',
          `/${USER_API}/1/Preferences/personal`,
          1,
        );
      });
    });
    describe('PERSON_SYNC', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[PERSON_SYNC]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[PERSON_SYNC]({ id: 1, data: { name: 'abcd' } });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'patch',
          `/${USER_API}/1/firstTimeSetup`,
          { name: 'abcd' },
        );
      });
    });

    describe('SIGNIN', () => {
      it('should call fetchWithAuthorisation', () => {
        expect(CONFIG.requests[SIGN_IN]).toEqual(AwsApi.signIn);
      });
    });
    describe('SIGNOUT', () => {
      it('should call fetchWithAuthorisation', () => {
        expect(CONFIG.requests[SIGN_OUT]).toEqual(AwsApi.signOut);
      });
    });
    describe('resendSignUp', () => {
      it('should call fetchWithAuthorisation', () => {
        expect(CONFIG.requests[RESEND_SIGNUP]).toEqual(AwsApi.resendSignUp);
      });
    });
    describe('resetPassword', () => {
      it('should call fetchWithAuthorisation', () => {
        expect(CONFIG.requests[RESET_PWD]).toEqual(AwsApi.resetPassword);
      });
    });

    describe('forgetPassword', () => {
      it('should call fetchWithAuthorisation', () => {
        expect(CONFIG.requests[FORGET_PWD]).toEqual(AwsApi.forgetPassword);
      });
    });

    describe('GET_ROLES', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[GET_ROLES]).toBe('function');
      });
      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[GET_ROLES]({ userId: 1 });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/1/roles`,
        );
      });
    });

    describe('GET_RECENT_ACTIVITY', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[GET_RECENT_ACTIVITY]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[GET_RECENT_ACTIVITY]('abcd@email.com');

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/me/activities`,
        );
      });
    });

    describe('HIDE_RECENT_ACTIVITY', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[HIDE_RECENT_ACTIVITY]).toBe('function');
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[HIDE_RECENT_ACTIVITY]({ nodeId: 1 });

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'patch',
          `/${USER_API}/${HIDE_RECENT_ACTIVITY}/1`,
        );
      });
    });

    describe('GET_ORGANISATION_INVITATIONS', () => {
      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();

        CONFIG.requests[GET_ORGANISATION_INVITATIONS]();

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/me/orgInvitations`,
        );
      });
    });

    describe('GET_USER_NODE', () => {
      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();

        CONFIG.requests[GET_USER_NODE]();

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/me/userNode`,
        );
      });
    });

    describe('GET_USER_RELATED_TEMPLATES', () => {
      it('should exists', () => {
        expect(typeof CONFIG.requests[GET_USER_RELATED_TEMPLATES]).toBe(
          'function',
        );
      });

      it('should call fetchWithAuthorisation', () => {
        requests.fetchWithAuthorisation = jest.fn();
        CONFIG.requests[GET_USER_RELATED_TEMPLATES]();

        expect(requests.fetchWithAuthorisation).toBeCalledWith(
          'get',
          `/${USER_API}/me/templates`,
        );
      });
    });

    describe('processError', () => {
      it('SIGN_IN', () => {
        expect(CONFIG.processError[SIGN_IN]()).toEqual({
          error: undefined,
          msg: undefined,
        });
      });
      it('CREATE_ORG_USER', () => {
        expect(CONFIG.processError[CREATE_ORG_USER]()).toEqual({
          error: undefined,
          msg: undefined,
        });
      });
      it('FORGET_PWD', () => {
        expect(CONFIG.processError[FORGET_PWD]()).toEqual({
          error: undefined,
          msg: undefined,
        });
      });
      it('RESET_PWD', () => {
        expect(CONFIG.processError[RESET_PWD]()).toEqual({
          error: undefined,
          msg: undefined,
        });
      });
      it('CHANGE_PWD', () => {
        expect(CONFIG.processError[CHANGE_PWD]()).toEqual({
          error: undefined,
          msg: undefined,
        });
      });
      it('CHANGE_PWD', () => {
        expect(CONFIG.processError[CREATE_USER_VIA_INVITE]()).toEqual({
          error: undefined,
          msg: undefined,
        });
      });
      it('CREATE_ORG_USER', () => {
        expect(CONFIG.processError[CREATE_PERSONAL_REGISTRATION]()).toEqual({
          error: undefined,
          msg: undefined,
        });
      });
    });
  });
});
