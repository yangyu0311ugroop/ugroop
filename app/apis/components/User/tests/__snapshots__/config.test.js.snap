// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User/config.js #requests still matches snapshot 1`] = `
Array [
  Array [
    "get",
    "/Users/me/invitations?show=default&status=pending",
  ],
  Array [
    "get",
    "/Users/me/invitations?show=received&status=pending",
  ],
]
`;
